vizro-core/tests/unit/vizro/models/_components/form/test_dropdown.py

```diff
@@ -139,22 +138,57 @@ def test_set_action_via_validator(self, identity_action_function):
 class TestDropdownBuild:
     """Tests model build method."""
 
-    def test_dropdown_with_all_option(self):
-        dropdown = Dropdown(options=["A", "B", "C"], title="Title", id="dropdown_id").build()
+    @pytest.mark.parametrize(
+        "value, options, expected_checkbox_value, expected_options, expected_value",
+        [
+            (
+                ["A"],
+                ["A", "B", "C"],
+                False,
+                [{"label": "A", "value": "A"}, {"label": "B", "value": "B"}, {"label": "C", "value": "C"}],
+                ["A"],
+            ),
+            (
+                ["A", "B", "C"],
+                ["A", "B", "C"],
+                True,
+                [{"label": "A", "value": "A"}, {"label": "B", "value": "B"}, {"label": "C", "value": "C"}],
+                ["A", "B", "C"],
+            ),
+            (
+                None,
+                ["A", "B", "C"],
+                True,
+                [{"label": "A", "value": "A"}, {"label": "B", "value": "B"}, {"label": "C", "value": "C"}],
+                ["A", "B", "C"],
+            ),
+        ],
+    )
+    def test_dropdown_with_all_option(self, value, options, expected_checkbox_value, expected_options, expected_value):
```

Comment:
Let's swap the order of `expected_options` and `expected_value` so it's aligned with the `test_checklist_build`.
```suggestion
    def test_dropdown_with_all_option(self, value, options, expected_checkbox_value, expected_value, expected_options):
```